tfs/src/main/java/hudson/plugins/tfs/rm/ConnectReleaseWebHookEvent.java

```diff
@@ -0,0 +1,319 @@
+            return new ConnectReleaseWebHookEvent();
+        }
+
+        @Override
+        public String getSampleRequestPayload() {
+            return "{\n"
+                 + "    \"eventType\": rmwebhook-create\n"
+                 + "    \"resource\": {"
+                 + "       \"webhookName\": \"webhook name\"\n"
+                 + "       \"payloadUrl\": \"https://xplatalm.vsrm.visualstudio.com/_apis/Release/receiveExternalEvent/wenhookId\"\n"
+                 + "       \"secret\": \"secret\"\n"
+                 + "     }"
+                 + "}";
+        }
+    }
+
+    @Override
+    public JSONObject perform(final ObjectMapper mapper, final Event event, final String message, final String detailedMessage) {
+        final Object resource = event.getResource();
+        final ReleaseWebHookResource parameters = mapper.convertValue(resource, ReleaseWebHookResource.class);
+
+        if (event.getEventType().equalsIgnoreCase(CREATE_WEBHOOK_EVENTNAME)) {
+            createReleaseWebHook(parameters);
+        } else if (event.getEventType().equalsIgnoreCase(REMOVE_WEBHOOK_ENENTNAME)) {
+            deleteReleaseWebHook(parameters);
+        } else if (event.getEventType().equalsIgnoreCase(LIST_WEBHOOK_EVENTNAME)) {
+            return listReleaseWebHook();
+        } else if (event.getEventType().equalsIgnoreCase(LINK_WEBHOOK_EVENTNAME)) {
+            linkWebHook(parameters);
+        } else if (event.getEventType().equalsIgnoreCase(UNLINK_WEBHOOK_EVENTNAME)) {
+            unlinkWebHook(parameters);
+        } else {
+            throw new UnsupportedOperationException("Webhook operation " + parameters.getOperationType() + " is not supported");
+        }
+
+        return JSONObject.fromObject(event);
+    }
+
+    private String validateAndGetPayloadUrl(final ReleaseWebHookResource parameters) {
+        String payloadUrl = parameters.getPayloadUrl();
+
+        if (StringUtils.isBlank(payloadUrl)) {
+            throw new InvalidParameterException("pyaloadUrl is empty");
+        }
+
+        final URI uri;
+        try {
+            uri = new URI(payloadUrl);
+        } catch (final URISyntaxException e) {
+            throw new InvalidParameterException("Malformed Payload URL " + e.getMessage());
+        }
+
+        final String hostName = uri.getHost();
+        if (StringUtils.isBlank(hostName)) {
+            throw new InvalidParameterException("Malformed Payload URL");
+        }
+
+        return StringUtils.stripEnd(StringUtils.trim(payloadUrl), "/");
+    }
+
+    private AbstractProject validateAndGetJenkinsProject(final ReleaseWebHookResource resource) {
+        if (StringUtils.isEmpty(resource.getProjectName())) {
+            throw new InvalidParameterException("Project name is empty");
+        }
+
+        for (final Item project : Jenkins.getActiveInstance().getAllItems()) {
+            if (project instanceof AbstractProject && project.getName().equalsIgnoreCase(resource.getProjectName())) {
```

Comment:
store result of resource.getProjectName locally rather than computing inside loop?

Reply:
fixed.